src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorParsingPhase.cs

```diff
@@ -18,7 +18,7 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
     {
         var options = codeDocument.GetParserOptions() ?? _optionsFeature.GetOptions();
         var syntaxTree = RazorSyntaxTree.Parse(codeDocument.Source, options);
-        codeDocument.SetSyntaxTree(syntaxTree);
+        codeDocument.SetPreTagHelperSyntaxTree(syntaxTree);
```

Comment:
From offline discussion, we can write a test that captures the new phase dependencies